pkg/controller/operator/seed/resources/kubermatic/kubermatic.go

```diff
@@ -84,10 +88,18 @@ func BackupContainersConfigMapCreator(cfg *operatorv1alpha1.KubermaticConfigurat
 
 			c.Data[storeContainerKey] = cfg.Spec.SeedController.BackupStoreContainer
 			c.Data[cleanupContainerKey] = cfg.Spec.SeedController.BackupCleanupContainer
-			if cfg.Spec.SeedController.BackupRestore.Enabled {
-				c.Data[deleteContainerKey] = cfg.Spec.SeedController.BackupDeleteContainer
+
+			if cfg.Spec.SeedController.BackupStoreContainer == strings.TrimSpace(common.DefaultBackupStoreContainer) &&
+				seed.Spec.BackupRestore != nil {
+				c.Data[storeContainerKey] = strings.TrimSpace(common.DefaultNewBackupStoreContainer)
+			}
+
+			if cfg.Spec.SeedController.BackupDeleteContainer == "" {
```

Comment:
Should we include this in the `seed.Spec.BackupRestore != nil` as well? also line 102?

Reply:
Why? it could be that the backup and restore has been enabled in the KubermaticConfigurations crd and was left out from the seed.

Reply:
I made a mistake there, I am sorry. The `BackupStoreContainer` is used in both legacy backup and BackupRestore controllers (with slightly different configuration). The `BackupDeleteCintainer` is _only_ used in the BackupRestore controllers. 

Defaulting logic should respect that. So, `BackupDeleteContainer` should be only enabled when `seed.Spec.BackupRestore != nil`.

Reply:
@moelsayed ok understood.